packages/framework/src/Component/AddSentryContextSubscriber.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component;
+
+use App\Environment;
+use RedisException;
+use Sentry\State\Scope;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Component\Domain\Exception\NoDomainSelectedException;
+use Shopsys\FrameworkBundle\Component\Localization\DisplayTimeZoneProviderInterface;
+use Shopsys\FrameworkBundle\Component\Maintenance\MaintenanceModeSubscriber;
+use Shopsys\FrameworkBundle\Component\Redis\RedisClientFacade;
+use Shopsys\FrameworkBundle\Component\Redis\RedisFacade;
+use Shopsys\FrameworkBundle\ShopsysFrameworkBundle;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Filesystem\Exception\FileNotFoundException;
+use Symfony\Component\HttpKernel\KernelEvents;
+use function Sentry\configureScope;
+
+class AddSentryContextSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Redis\RedisClientFacade $redisClientFacade
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Shopsys\FrameworkBundle\Component\Localization\DisplayTimeZoneProviderInterface $displayTimeZoneProvider
+     * @param \Shopsys\FrameworkBundle\Component\Redis\RedisFacade $redisFacade
+     */
+    public function __construct(
+        protected readonly RedisClientFacade $redisClientFacade,
+        protected readonly Domain $domain,
+        protected readonly DisplayTimeZoneProviderInterface $displayTimeZoneProvider,
+        protected readonly RedisFacade $redisFacade,
+    ) {
+    }
+
+    public function setSentryContext(): void
+    {
+        $context = [
+            'environment' => class_exists('App\Environment') ? Environment::getEnvironment() : '',
+            'version' => ShopsysFrameworkBundle::VERSION,
+            'systemTimeZone' => date_default_timezone_get(),
+        ];
+
+        try {
+            $this->redisFacade->pingAllClients();
```

Comment:
Why is this call needed? this is a heavy operation and it can be performance problem :)

Reply:
is it heavy indeed? I thought the ping is quite a lightweight operation...Do you have a better suggestion how to check whether the redis is available? The reason is that this subscriber is called on console command, and one command is called during the image build as a composer post install script (`shopsys:domains-urls:configure`). In that case, Redis is not available hence you are not able to check whether the maintenance mode is enabled :grimacing:

Reply:
ah, maybe we can remove this line as `$this->redisClientFacade->contains` should also throw `RedisException` if redis is not available :thinking: :slightly_smiling_face:

Reply:
Yes, it's a heavy operation. It opens ~10 (It depends on how many it was already connected during request) connections to Redis and every connection is time consuming